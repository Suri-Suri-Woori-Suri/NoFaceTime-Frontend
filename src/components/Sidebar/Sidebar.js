import React from 'react';
import styles from './Sidebar.module.css';
import { Link } from 'react-router-dom';


const Sidebar = () => {
  return (
    <div className={styles.SideContainer}>
      <div className={styles.MenuBox}>
        <Link className={styles.Menu} to="/rooms">Room</Link>
        <Link className={styles.Menu} to="/groups">Group</Link>
      </div>
    </div>
  );
};

export default Sidebar;