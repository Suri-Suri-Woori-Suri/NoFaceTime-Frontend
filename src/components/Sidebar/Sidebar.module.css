.SideContainer {
  vertical-align: top;
  width: 20%;
  background: #eee;
  display: table-cell;
}

.MenuBox {
  margin-top: 80%;
  background-color: tomato;
}

.Menu {
  display: block;
  height: 40px;
  width: 100%;
  padding: 20px 0px;
  text-decoration: none;
  font-size: 25px;
  color: #2196F3;
  background-color: rgb(223, 243, 224);
  border: 1px solid rgba(0, 0, 0, 0.267);
}

.Menu:hover {
  background-color: rgb(183, 237, 250);
}
