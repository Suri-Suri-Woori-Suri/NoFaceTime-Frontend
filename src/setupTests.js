import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

export const renderWithRouter = component => render(<Router>{component}</Router>);
export const mountWithRouter = component => mount(<Router>{component}</Router>);

configure({ adapter: new Adapter() });
