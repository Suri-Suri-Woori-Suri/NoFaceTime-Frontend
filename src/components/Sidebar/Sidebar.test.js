import React from 'react';
import { cleanup } from '@testing-library/react';

import Sidebar from './Sidebar';
import { mountWithRouter, renderWithRouter } from '../../setupTests';

describe('<Sidebar/>', () => {
  afterEach(cleanup);

  const wrapper = mountWithRouter(<Sidebar />);

  it('should be render lists correctly', () => {
    const { getByText } = renderWithRouter(<Sidebar />);

    expect(getByText('Room')).toBeInTheDocument(true);
    expect(getByText('Group')).toBeInTheDocument(true);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
