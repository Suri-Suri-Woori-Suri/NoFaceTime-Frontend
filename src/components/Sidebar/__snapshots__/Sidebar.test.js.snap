// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Sidebar>
      <div
        className="SideBar"
      >
        <div
          className="MenuBox"
        >
          <Link
            className="Menu"
            to="/rooms"
          >
            <LinkAnchor
              className="Menu"
              href="/rooms"
              navigate={[Function]}
            >
              <a
                className="Menu"
                href="/rooms"
                onClick={[Function]}
              >
                Room
              </a>
            </LinkAnchor>
          </Link>
          <Link
            className="Menu"
            to="/groups"
          >
            <LinkAnchor
              className="Menu"
              href="/groups"
              navigate={[Function]}
            >
              <a
                className="Menu"
                href="/groups"
                onClick={[Function]}
              >
                Group
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </div>
    </Sidebar>
  </Router>
</BrowserRouter>
`;
